app/ports/models/port.py

```diff
@@ -107,32 +113,31 @@ def load_ports_and_maintainers_table(ports):
                 except KeyError:
                     pass
 
+        @transaction.atomic
         def load_dependencies_table(ports):
-
-            def load_depends(list_of_dependencies, type_of_dependency, port):
+            def load_depends(list_of_dependencies, type_of_dependency, pid):
                 dependency = Dependency()
                 dependencies = []
                 dependency.type = type_of_dependency
-                dependency.port_name = port
+                dependency.port_name_id = pid
```

Comment:
`port_name` is the name of the field which is a foreign key. The actual field is thus `port_name_id` which holds the id of the foreign object.

I am not sure if I got your point right?

Reply:
Actually, I am not sure about `port_name_id` either. There could be reasons behind it because of the strict Django conventions but does it get linked to the field `port_name` using `port_name_id` above?